### PR TITLE
Revisit composedPath update

### PR DESCRIPTION
**Pull request by UncleGedd**
This PR revisits #323 to discuss lingering talking points including:
- Testing strategy for `buildPath`
- Deprecating `initEvent`
- Refactor `buildPath` how @Jyyjy originally intended

### CODE REVIEW

**Comment by UncleGedd** [comment]
Made 2 updates thus far:
1. Refactored the `buildPath` function how @Jyyjy orignally intended ([reference](https://github.com/apache/incubator-flagon-useralejs/pull/323#issuecomment-1365557602)) including fixing the unit test
2. Deleted the `defaults to path if available` unit test. Someone check me on this, but I'm not sure this test is relevant anymore

File: test/packageLogs_spec.js [hunk 1]
```diff
@@ -313,28 +313,15 @@ describe('packageLogs', () => {
...
-            const evt = document.createEvent('CustomEvent');
-            document.body.appendChild(ele);
-            evt.initEvent('testEvent', true, true);
-            ele.dispatchEvent(evt);
-            evt.composedPath = function() {
-                let ele = evt.target;
-                return [ele, ele.parentElement, ele.parentElement.parentElement];
-            };
-            expect(buildPath(evt)).to.deep.equal(['div', 'body', 'html']);
+            const expectedPath = ['div', 'body', 'html',   "#document", "Window"]
```
Comment (UncleGedd) [inline test/packageLogs_spec.js:323]: Noting here that the `expectedPath` now includes `"#document"` and `"Window"`. This is expected behavior with `composedPath`, see screenshot from Chrome dev tools below:
Screen Shot 2023-01-02 at 11 14 13 AM

**Comment by UncleGedd** [comment]
Ready for review! I added a Cypress test as well to test the `buildPath` behavior in a real browser. Also, refactored an old test that wasn't working properly (`"does not send logs on page exit when config is off"`)

**Comment by Jyyjy** [comment]
These are the changes that I intended. Thanks @UncleGedd!

**Comment by UncleGedd** [comment]
@poorejc I forget our process. Do you want to merge this into `test` first or go directly into `master`?

File: test/packageLogs_spec.js (hunk 1 above)
Comment (poorejc) [reply test/packageLogs_spec.js]: @UncleGedd AWESOME. Thanks, Gedd. Was planning some maintenance next week... I'll beat on this a bit and report back!

**Comment by poorejc** [comment]
@UncleGedd this is a really elegant update--I'm in love with the Cypress tests! Very nice. Tests well. Merged.

**Pull request merged**